src/oneelement.jl

```diff
@@ -0,0 +1,27 @@
+"""
+    OneElement(val, ind, axes) <: AbstractArray
+Extremely simple `struct` used for the gradient of scalar `getindex`.
```

Comment:
It's possible that this should have friendlier constructors, which e.g. call `to_indices` to convert any other index input, and allow you to supply size not axes? 

Or it could just mention that `setindex(Zeros(), ...)` is the intended interface.

Reply:
This was already added, I've added more docs.